syntax = "proto3";

message MsgRequest {
    int32 id = 1;
    string interface = 2;
    int32 customer_request_id = 3;
    int32 logical_clock = 4; // Add a logical clock field
    repeated CustomerRequestMap branch_to_customer_req_id_map = 5;

}

message CustomerRequestMap {
    int32 branch_id = 1;
    repeated int32 corresponding_customer_request_id = 2;
}

message MsgResponse {
    string interface = 1;
    int32 customer_request_id = 2;
    int32 logical_clock = 3; // Add a logical clock field
}

service Branch {
    rpc MsgDelivery(MsgRequest) returns (MsgResponse) {}
    rpc MsgPropagation(MsgRequest) returns (MsgResponse) {}
}
